compiler/modules/nback/structure.bal

```diff
@@ -67,6 +67,78 @@ final RuntimeFunction mappingConstructFunction = {
 
 const LLVM_INDEX = "i32";
 
+// This is used in the name of the runtime function.
+enum ListReprKind {
+    LIST_GENERIC = "generic",
+    LIST_INT_ARRAY = "int_array",
+    LIST_FLOAT_ARRAY = "float_array"
+}
+
+type ListRepr readonly & object {
+    ListReprKind kind;
+    llvm:Type memberType;
+    int listDescGetIndex;
+    int listDescSetIndex;
+    function buildMember(llvm:Builder builder, Scaffold scaffold, bir:Operand member, t:SemType memberType) returns llvm:Value|BuildError;
+    function buildMemberStore(llvm:Builder builder, Scaffold scaffold, llvm:Value value, bir:Register reg);
+};
+
+readonly class GenericListRepr {
+    *ListRepr;
+    ListReprKind kind = LIST_GENERIC;
+    llvm:Type memberType = LLVM_TAGGED_PTR;
+    int listDescGetIndex = 1;
+    int listDescSetIndex = 2;
+    function buildMember(llvm:Builder builder, Scaffold scaffold, bir:Operand member, t:SemType memberType) returns llvm:Value|BuildError {
+        return buildWideRepr(builder, scaffold, member, REPR_ANY, memberType);
+    }
+    function buildMemberStore(llvm:Builder builder, Scaffold scaffold, llvm:Value value, bir:Register reg) {
+        return buildStoreTagged(builder, scaffold, value, reg);
+    }
+}
+
+readonly class IntListRepr {
+    *ListRepr;
+    ListReprKind kind = LIST_INT_ARRAY;
+    llvm:Type memberType = LLVM_INT;
+    int listDescGetIndex = 3;
+    int listDescSetIndex = 4;
+    function buildMember(llvm:Builder builder, Scaffold scaffold, bir:Operand member, t:SemType memberType) returns llvm:Value|BuildError {
+        return buildInt(builder, scaffold, <bir:IntOperand>member);
+    }
+    function buildMemberStore(llvm:Builder builder, Scaffold scaffold, llvm:Value value, bir:Register reg) {
+        return buildStoreInt(builder, scaffold, value, reg);
+    }
+}
+
+readonly class FloatListRepr {
+    *ListRepr;
+    ListReprKind kind = LIST_FLOAT_ARRAY;
+    llvm:Type memberType = LLVM_DOUBLE;
+    int listDescGetIndex = 5;
+    int listDescSetIndex = 6;
+    function buildMember(llvm:Builder builder, Scaffold scaffold, bir:Operand member, t:SemType memberType) returns llvm:Value|BuildError {
+        return buildFloat(builder, scaffold, <bir:FloatOperand>member);
+    }
+    function buildMemberStore(llvm:Builder builder, Scaffold scaffold, llvm:Value value, bir:Register reg) {
+        return buildStoreFloat(builder, scaffold, value, reg);
+    }
+}
+
+final ListRepr LIST_GENERIC_REPR = new GenericListRepr();
```

Comment:
We generally use all caps only for const (or things that would be const except for jBallerina/spec bugs).